### PR TITLE
Fix FileDialog default size

### LINKED ISSUE #58780: FileDialog appears super-small
### Godot version

cdd63fa

### System information

W10

### Issue description

```GDScript
$FileDialog.popup_centered()
```
image
Default size is probably too small.

### Steps to reproduce

1. Add FileDialog
2. Popup it

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Giganzo**
Fixes: https://github.com/godotengine/godot/issues/58780

Before: 
Screenshot_20240914_132037

After:
Screenshot_20240914_161601

### CODE REVIEW

File: scene/gui/file_dialog.cpp [hunk 1]
```diff
@@ -1526,6 +1526,7 @@ FileDialog::FileDialog() {
 	update_dir();
 
 	set_hide_on_ok(false);
+	set_size(Size2(640, 360));
```
Comment (AThousandShips) [inline scene/gui/file_dialog.cpp:1529]: This seems like a very brute force solution to me, and ignores things like scaling, I think the solution is elsewhere and that the size isn't computed correctly from the contents

Comment (KoBeWi) [reply scene/gui/file_dialog.cpp:1529]: The size is computed correctly. It's the minimum size, which is used if you don't override it.

**Comment by WhalesState** [comment]
This change should be added to the ProjectSettings.xml to paypass the checks,
You can open the failed check and copy the line to the same place in the XML file.

The proper way is to use the --doctool from cmd.

`path/to/godot-source-code/bin/./godot.exec --doctool path/to/godot-source-code/`

If you are using VSCode, you can edit and use this build task.

```json
{
    "label": "build docs",
    "group": "build",
    "type": "shell",
    "command": "${workspaceFolder}/bin/./godot.windows.editor.x86_64.exe --doctool ${workspaceFolder}/"
}
```

**Review by KoBeWi — APPROVED** [review]
The new default size makes the dialog much more usable. I tested and it works correctly with scaling enabled too.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**